hkust-taco/mlscript
# Add `driver` subproject

- Implement multiple-file compilation
- Add `import` and `export` keywords
- Update `ts2mls` DiffTests and test files
- Import `libes5.d.ts` and cherry-pick from `dom.d.ts`
- Add `UnusableLike` and `Unsupported` types
- Add ident escaping
- Implement invocation of ts libraries in mlscript
- Allow overriding built-in types from `Predef.mlsi`
- Optimize `NewParser.block` with tail-recursion
- Fix literal typing
- Handle selections in inheritance
- Update the implementation check for `declare` keyword
- Enhance `ts2mls`:
  - Mark complex signatures as unsupported
  - Fix generation of overloading, optional parameters and var-length parameters
  - Support `import`/`require` and `export` parsing
  - Support `export as namespace` and triple slashes
  - Fix flag checking
  - Fix private members' name
